src/contexts/Config/Config.tsx

```diff
@@ -0,0 +1,25 @@
+import { createContext, FunctionComponent, useContext, useEffect } from "react";
+import { fetchConfig, Config } from "../../api/config";
+import { useRequest, UseRequestResponse } from "../../hooks/useRequest";
+
+const ConfigContext = createContext<UseRequestResponse<Config>>({
+  loading: false,
+  data: undefined,
+  error: undefined,
+});
+
+export const useConfig = () => useContext(ConfigContext);
+
+export const ConfigProvider: FunctionComponent = ({ children }) => {
+  const [requestConfig, configResponse] = useRequest(fetchConfig);
+
+  useEffect(() => {
+    void requestConfig();
+  }, [requestConfig]);
+
+  return (
+    <ConfigContext.Provider value={configResponse}>
+      {children}
```

Comment:
I'd probably suggest rendering an app-level loader here if the config response hasn't resolved

```tsx
{configResponse.data ? children : <Loader />}
```

Reply:
I think components that need the config should be responsible for showing loaders at lower levels if need be, I don't see the need to now show the whole app if config isn't loaded. That being said, I need to revise `null` rendering in the `PackageDetails` component.